clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy

```diff
@@ -425,7 +415,13 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
         }
         namedPorts = sourceServerGroup?.namedPorts?.collect { name, port -> new NamedPort(name: name, port: port) }
       } else {
-        namedPorts = description?.loadBalancingPolicy?.namedPorts
+        def loadBalancingPolicy = description?.loadBalancingPolicy
+        if (loadBalancingPolicy?.namedPorts)  {
```

Comment:
If someone sets `namedPorts` to an empty list (that would be strange), and they haven't removed `loadBalancingPolicy.listeningPort` from their deploy description, they'll hit the `else if` below, won't they? That would be unexpected, though maybe unlikely. I would do a strict null check here anyways.